indico/modules/search/client/js/components/SearchApp.jsx

```diff
@@ -106,51 +88,87 @@ SearchTypeMenuItem.defaultProps = {
   onClick: undefined,
 };
 
+function useQueryParams() {
+  const [query, _setQuery] = useState(window.location.search);
+  const queryObject = useMemo(() => Object.fromEntries(new URLSearchParams(query)), [query]);
+  const history = useHistory();
+
+  const setQuery = useCallback(
+    (type, name, reset) => {
+      const params = new URLSearchParams(reset ? undefined : query);
+      if (name !== undefined) {
+        params.set(type, name);
+      } else {
+        params.delete(type);
+      }
+      const _query = params.toString();
+      history.push({
+        pathname: window.location.pathname,
+        search: `?${_query}`,
+      });
+      _setQuery(_query);
+    },
+    [history, query]
+  );
+
+  useEffect(() => {
+    history.listen(location => {
```

Comment:
this listener should probably be removed in the useEffect's cleanup function

Reply:
Fair point!